include/st_formatter.h

```diff
@@ -625,6 +625,31 @@ namespace ST
     }
 #endif
 
+#ifdef ST_ENABLE_COREFOUNDATION
+    inline void format_type(const ST::format_spec &format, ST::format_writer &output,
+                            const CFStringRef str)
+    {
+        size_t size = (size_t)CFStringGetLength(str);
+        const char* cstr = CFStringGetCStringPtr(str, kCFStringEncodingUTF8);
+        if (cstr) {
+            ST::format_string(format, output, cstr, size);
+        } else {
+            char* buf = new char[size+1];
+            CFStringGetCString(str, buf, size+1, kCFStringEncodingUTF8);
+            ST::format_string(format, output, buf, size);
+            delete[] buf;
```

Comment:
You can use an `ST::char_buffer` to avoid manual allocation/deallocation here.  (Note that `allocate()` already reserves one extra byte for the nul terminator)
```suggestion
            ST::char_buffer buf;
            buf.allocate(size);
            CFStringGetCString(str, buf.data(), size+1, kCFStringEncodingUTF8);
            ST::format_string(format, output, buf.data(), size);
```